tests/framework/test_time.py

```diff
@@ -80,26 +87,35 @@ def get_index_by_parity(index, parity):
         return index
 
 
-def get_pop_by_parity(sim, parity):
+def get_pop_by_parity(sim: SimulationContext, parity: str) -> pd.DataFrame:
     pop = sim.get_population()
     return pop.loc[get_index_by_parity(pop.index, parity)]
 
 
-def pipeline_by_parity(sim, step_modifiers, parity):
+def pipeline_by_parity(
+    sim: SimulationContext, step_modifiers: dict[str, int], parity: str
+) -> pd.Series[Any]:
```

Comment:
These arent' just all floats?

Reply:
These are the different clock step types and using Any worked instead of ignoring it.